app/models/user.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+class User < ApplicationRecord
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable,
+         :omniauthable, omniauth_providers: [:google_oauth2]
+
+  def self.from_omniauth(auth)
+    find_existing_user(auth) || create_new_user(auth)
+  end
+
+  private_class_method def self.find_existing_user(auth)
```

Comment:
The team never use `private_class_method` and I have not seen it used in the Ruby community either 😅 Why use that syntax?

Reply:
@olivierobert Since we have `from_omniauth()` as the self methods so that's why I added `private_class_method` to make the `find_existing_user` and `create_new_user` can be invoked 😅 I updated to use `class << self` with `private` section instead 🙏